plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx

```diff
@@ -85,7 +86,10 @@ const ProductCollectionInspectorControls = (
 	const showFilterableControl =
 		! isArchiveTemplate && shouldShowFilter( CoreFilterNames.FILTERABLE );
 	const showOrderControl =
-		showQueryControls && shouldShowFilter( CoreFilterNames.ORDER );
+		showQueryControls &&
+		shouldShowFilter( CoreFilterNames.ORDER ) &&
+		// See https://github.com/woocommerce/woocommerce/issues/50862
+		attributes.collection !== 'woocommerce/product-collection/hand-picked';
```

Comment:
Instead of this specific check, please add `CoreFilterNames.ORDER` to `hideControls` in a collection definition.